benchmark-framework/src/main/java/io/openmessaging/benchmark/worker/BenchmarkWorker.java

```diff
@@ -64,24 +64,33 @@ public static void main(String[] args) throws Exception {
             System.exit(-1);
         }
 
-        Configuration conf = new CompositeConfiguration();
-        conf.setProperty(Stats.STATS_PROVIDER_CLASS, PrometheusMetricsProvider.class.getName());
-        conf.setProperty("prometheusStatsHttpPort", arguments.statsPort);
-        Stats.loadStatsProvider(conf);
-        StatsProvider provider = Stats.get();
-        provider.start(conf);
-
-        Runtime.getRuntime().addShutdownHook(new Thread(
-            () -> provider.stop(),
-            "benchmark-worker-shutdown-thread"));
-
-        // Dump configuration variables
-        log.info("Starting benchmark with config: {}", writer.writeValueAsString(arguments));
-
-        // Start web server
-        Javalin app = Javalin.start(arguments.httpPort);
+        try {
 
-        new WorkerHandler(app, provider.getStatsLogger("benchmark"));
+            Configuration conf = new CompositeConfiguration();
+            conf.setProperty(Stats.STATS_PROVIDER_CLASS, PrometheusMetricsProvider.class.getName());
+            conf.setProperty("prometheusStatsHttpPort", arguments.statsPort);
+            Stats.loadStatsProvider(conf);
+            StatsProvider provider = Stats.get();
+            provider.start(conf);
+
+            Runtime.getRuntime().addShutdownHook(new Thread(
+                () -> provider.stop(),
+                "benchmark-worker-shutdown-thread"));
+
+            // Dump configuration variables
+            log.info("Starting benchmark with config: {}", writer.writeValueAsString(arguments));
+
+            // Start web server
+            Javalin app = Javalin.start(arguments.httpPort);
+
+            new WorkerHandler(app, provider.getStatsLogger("benchmark"));
+        } catch (Exception e) {
```

Comment:
Nit: Now that we're catching exceptions thrown in main do we need the `throws Exception` declaration in it's function signature anymore?

Reply:
We don't, let me remove it.

Reply:
Turns out we do. This exception stuff was actually because it happens inside a lamba bassed to `Map.forEach` and _these_ can't throw exceptions. 

There is actually a bunch of stuff at the top level inside `main` not changed here, which throw checked exceptions, so we still need `throws Exception`.

Reply:
It is removed in the current version.